### PR TITLE
Should collide

### LINKED ISSUE #411: Hitbox collides only with specific tags
## Describe the solution you'd like
I want a HitBox be able to interact with Hitboxes with only a specific tag.
ie. enemies can walk on clouds but not the player

## Describe alternatives you've considered
maybe in an on_collide you can just undo the collision. Or a proper way to do it.

### PR DESCRIPTION
**Pull request by tinmarr**
Closes #411 
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [x] Documentation is up to date
  - [x] Relevent tutorials are up to date
  - [ ] Changelog is up to date
  - [x] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*

### CODE REVIEW

File: rubato/structure/gameobject/physics/engine.py [hunk 1]
```diff
@@ -140,6 +140,9 @@ def collide(hitbox_a: Hitbox, hitbox_b: Hitbox) -> Optional[Manifold]:
         Returns:
             Returns a collision info object if a collision is detected or None if no collision is detected.
         """
+        if not hitbox_a.should_collide(hitbox_a, hitbox_b) or not hitbox_b.should_collide(hitbox_b, hitbox_a):
```
Comment (mrElnekave) [inline rubato/structure/gameobject/physics/engine.py:143]:
```suggestion
        if not (hitbox_a.should_collide(hitbox_a, hitbox_b) and hitbox_b.should_collide(hitbox_b, hitbox_a)):
```
sorry previous hurts ma brain

Comment (tinmarr) [reply rubato/structure/gameobject/physics/engine.py:143]: But it's less clear!!!

**Review by mrElnekave — DISMISSED** [review]
goog

File: rubato/structure/gameobject/physics/engine.py [hunk 2]
```diff
@@ -140,6 +140,9 @@ def collide(hitbox_a: Hitbox, hitbox_b: Hitbox) -> Optional[Manifold]:
...
+            return None
```
Comment (mrElnekave) [inline rubato/structure/gameobject/physics/engine.py:144]:
```suggestion
            return
```

**Review by mrElnekave — APPROVED** [review]

**Pull request merged**